src/openms/source/FORMAT/HANDLERS/MzIdentMLDOMHandler.cpp

```diff
@@ -1280,7 +1280,7 @@ namespace OpenMS
               {
                 pep_id_->push_back(PeptideIdentification());
                 pep_id_->back().setHigherScoreBetter(false); //either a q-value or an e-value, only if neither available there will be another
-                pep_id_->back().setMetaValue("spectrum_reference", spectrumID); // TODO @mths consider SpectrumIDFormat to get just a index number here
+                pep_id_->back().setMetaValue("spectrum_reference", spectrumID);  // as spectrumID is taken from the mz_file and varies widely from vendor to vendor, spectrum_reference as string will reference the spectrum internally
```

Comment:
the comment is a bit misleading. is says that spectrumID varies a lot, thus, we take the spectrum_reference. But in the actual code you set the spectrumID.